src/main/java/com/uid2/admin/managers/KeysetManager.java

```diff
@@ -0,0 +1,122 @@
+package com.uid2.admin.managers;
+
+import com.uid2.admin.secret.IKeysetKeyManager;
+import com.uid2.admin.store.writer.KeysetStoreWriter;
+import com.uid2.shared.Const;
+import com.uid2.shared.auth.ClientKey;
+import com.uid2.shared.auth.Keyset;
+import com.uid2.shared.auth.Role;
+import com.uid2.shared.store.reader.RotatingKeysetProvider;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Map;
+
+import static java.lang.Math.max;
+
+public class KeysetManager {
+
+    private final RotatingKeysetProvider keysetProvider;
+    private final KeysetStoreWriter keysetStoreWriter;
+    private final IKeysetKeyManager keysetKeyManager;
+
+    private final boolean enableKeysets;
+
+    public KeysetManager(RotatingKeysetProvider keysetProvider, KeysetStoreWriter keysetStoreWriter,
+                         IKeysetKeyManager keysetKeyManager, boolean enableKeysets) {
+        this.keysetProvider = keysetProvider;
+        this.keysetStoreWriter = keysetStoreWriter;
+        this.keysetKeyManager  = keysetKeyManager;
+        this.enableKeysets = enableKeysets;
+    }
+
+
+    public static final String MasterKeysetName = "Master";
+    public static final String RefreshKeysetName = "Refresh";
+    public static final String FallbackPublisherKeysetName = "Fallback Publisher";
+
+
+    public static Keyset lookUpKeyset(int siteId, Map<Integer, Keyset> keysets) {
+        for (Keyset keyset: keysets.values()) {
+            if(keyset.getSiteId() == siteId && keyset.isDefault()) {
+                return keyset;
+            }
+        }
+        return null;
+    }
+
+    public static Integer getMaxKeyset(Map<Integer, Keyset> keysets) {
```

Comment:
do we need this to be public? when should consumer call this vs getNextKeysetId ?

Reply:
Not at the moment but maybe after the keyset migration ticket